tests/ui_tests/app/page/cluster/page.py

```diff
@@ -312,3 +316,101 @@ def delete_host_by_row(self, row: WebElement):
         self.wait_element_visible(DeleteDialog.body)
         self.find_and_click(DeleteDialog.yes)
         self.wait_element_hide(DeleteDialog.body)
+
+
+@dataclass
+class ComponentsHostRowInfo:
+    """Information from host row about host on Components page"""
+
+    name: str
+    components: str
+
+
+class ClusterComponentsPage(ClusterPageMixin):
+    """Cluster page components menu"""
+
+    MENU_SUFFIX = 'host_component'
+    MAIN_ELEMENTS = [
+        ObjectPageLocators.title,
+        ObjectPageLocators.subtitle,
+        ClusterComponentsLocators.restore_btn,
+        ClusterComponentsLocators.save_btn,
+        ClusterComponentsLocators.components_title,
+        ClusterComponentsLocators.hosts_title,
+        ClusterComponentsLocators.service_page_link,
+        ClusterComponentsLocators.hosts_page_link,
+    ]
+
+    def click_service_page_link(self):
+        self.find_and_click(ClusterComponentsLocators.service_page_link)
+
+    def click_hosts_page_link(self):
+        self.find_and_click(ClusterComponentsLocators.hosts_page_link)
+
+    def click_add_host_btn(self):
+        self.find_and_click(ClusterComponentsLocators.create_hosts_btn)
+        self.wait_element_visible(HostCreationLocators.block)
+
+    @allure.step("Get all host rows")
+    def get_host_rows(self):
+        return self.find_elements(ClusterComponentsLocators.host_row)
+
+    @allure.step("Get all components rows")
+    def get_components_rows(self):
+        return self.find_elements(ClusterComponentsLocators.component_row)
+
+    @allure.step("Get info by row")
+    def get_row_info(self, row: WebElement):
+        return ComponentsHostRowInfo(
+            name=self.find_child(row, ClusterComponentsLocators.Row.name).text,
+            components=self.find_child(row, ClusterComponentsLocators.Row.number).text,
+        )
+
+    def find_host_row_by_name(self, host_name: str):
+        for host in self.get_host_rows():
+            host_name_loc = self.find_child(host, ClusterComponentsLocators.Row.name)
```

Comment:
here and below it's unclear what does "loc" suffix mean
maybe it's `host_name_element` or smt
and maybe it's better to be `host_row` instead of `host` as the last is for "client" `Host` objects

Reply:
renamed, thnx)